pkg/volume/csi/csi_plugin.go

```diff
@@ -324,24 +371,44 @@ func (p *csiPlugin) ConstructVolumeSpec(volumeName, mountPath string) (*volume.S
 	}
 
 	klog.V(4).Info(log("plugin.ConstructVolumeSpec extracted [%#v]", volData))
-
-	fsMode := api.PersistentVolumeFilesystem
-	pv := &api.PersistentVolume{
-		ObjectMeta: meta.ObjectMeta{
-			Name: volData[volDataKey.specVolID],
-		},
-		Spec: api.PersistentVolumeSpec{
-			PersistentVolumeSource: api.PersistentVolumeSource{
-				CSI: &api.CSIPersistentVolumeSource{
-					Driver:       volData[volDataKey.driverName],
-					VolumeHandle: volData[volDataKey.volHandle],
+	var spec *volume.Spec
+
+	if p.inlineFeatureEnabled && volData[volDataKey.sourceKind] == volumeKind {
+		handle := volData[volDataKey.volHandle]
```

Comment:
unclear to me why we're using this `volData` map instead of passing variables around. This seems much more brittle, especially since nowhere in this function do we check for the values in the map already existing.

If we MUST use the map, we should do nil checks and fail fast if the values are not there